### PR TITLE
feat: disruption approval email

### PR DESCRIPTION
**Pull request by DeannaSharma**
- created disruption approval email (the email should only contain a link to the disruption, you will not be able to approve or reject the disruption via the email)
- current approval emails will always sent to org admin - logic to send only to org admins that opt in to email notifications will be implemented in ticket 476
- updated publish and publish-edit api so that when a disruption is submitted by a staff a member the approval email is sent to the staff users organisation admins
- if no org admin exists the disruption is still submitted and the staff user will be redirected to the dashboard page, a logger error will be created.
- updated the feedback emailer to use eu-west-2 SES identity rather than us-east-1, updated feedback api tests accordingly

To test set up a staff and org admin account. Add the email address associated with the org admin account as a verified identity in SES. Log into your staff account and create a disruption. Go to your org admin accounts email, open the approval email link in an incognito tab and log in to your org admin account to approve or reject the disruption. Repeat for editing a disruption.

### CODE REVIEW

File: site/pages/api/publish-edit.api.ts
Comment (alvak1) [inline site/pages/api/publish-edit.api.ts:172]: Conscious that with increasing steps in publish flow - the response returned to the user could be delayed. Can we refactor the above block of code and invoke it asynchronously?

File: site/pages/api/publish.api.ts
Comment (alvak1) [inline site/pages/api/publish.api.ts:112]: Could we refactor this to a function that can be re-used in publish-edit flow?

File: site/pages/api/publish-edit.test.ts
Comment (alvak1) [inline site/pages/api/publish-edit.test.ts:342]: If an org admin is not yet setup - should the disruption be impacted? The disruption can be in pending to approval status which can be approved by a publisher or approved later when an admin user is created?

Comment (alvak1) [inline site/pages/api/publish-edit.test.ts:342]:
```suggestion
    it("should redirect to dashboard and log an error if an org staff creates a disruption and no org admins exists for staff member's org", async () => {
```

File: site/pages/api/publish.test.ts
Comment (alvak1) [inline site/pages/api/publish.test.ts:220]: Should we be explicit in returning a list without admins and verifying that the function `getOrgAdminEmailsForStaffUserOrg` was invoked using a spy?

File: site/pages/api/publish-edit.test.ts
Comment (DeannaSharma) [reply site/pages/api/publish-edit.test.ts:342]: done

Comment (DeannaSharma) [reply site/pages/api/publish-edit.test.ts:342]: The disruption isn't impacted. If an org admin doesn't exist the disruption will be published still but an email won't be sent. We only log an error message to help with troubleshooting if needed.

File: site/pages/api/publish-edit.api.ts
Comment (DeannaSharma) [reply site/pages/api/publish-edit.api.ts]: refactored to create a separate function sendDisruptionApprovalEmail() that is invoked asynchronously

File: site/pages/api/publish.api.ts
Comment (DeannaSharma) [reply site/pages/api/publish.api.ts:107]: refactored to create a separate function sendDisruptionApprovalEmail()

File: site/pages/api/publish.test.ts
Comment (DeannaSharma) [reply site/pages/api/publish.test.ts]: This is now outdated, similar tests have been created for the ```sendDisruptionApprovalEmail``` function.

In this case, I don't think it's necessary to check if ```getOrgAdminEmailsForStaffUserOrg``` has been invoked since we mock the required inputs and are checking for the correct outputs. Although I have created unit tests for this function to ensure given certain inputs we get the expected outputs.

File: site/pages/api/publish.test.ts
Comment (alvak1) [inline site/pages/api/publish.test.ts:188]:
```suggestion
    it("should call sendDisruptionApprovalEmail method when an org staff creates a disruption", async () => {
```

File: site/utils/apiUtils/disruptionApprovalEmailer.ts
Comment (alvak1) [inline site/utils/apiUtils/disruptionApprovalEmailer.ts:97]: find or filter function can be used here to obtain the org admins based on staff org id if the data returned from `getAllUsersInGroup` is parsed by a schema. All data from external sources are parsed using zod.

File: site/pages/api/publish-edit.test.ts
Comment (DeannaSharma) [reply site/pages/api/publish-edit.test.ts:311]: done

File: site/utils/apiUtils/disruptionApprovalEmailer.ts
Comment (DeannaSharma) [reply site/utils/apiUtils/disruptionApprovalEmailer.ts]: refactored

**Review by alvak1 — APPROVED** [review]
Approved

**Pull request merged**
